include/seqan3/alphabet/structure/concept.hpp

```diff
@@ -267,46 +267,66 @@ namespace seqan3::detail::adl_only
-                                                 std::remove_cvref_t<alph_t>,
-                                                 std::type_identity<alph_t>>>
-struct max_pseudoknot_depth_fn
+template <typename alphabet_t>
+struct max_pseudoknot_depth_cpo : public detail::customisation_point_object<max_pseudoknot_depth_cpo<alphabet_t>, 2>
 {
-public:
-    SEQAN3_CPO_IMPL(2, (deferred_type_t<seqan3::custom::alphabet<alph_t>, decltype(v)>::max_pseudoknot_depth)) // custom
-    SEQAN3_CPO_IMPL(1, (max_pseudoknot_depth(v)                                                             )) // ADL
-    SEQAN3_CPO_IMPL(0, (deferred_type_t<std::remove_cvref_t<alph_t>, decltype(v)>::max_pseudoknot_depth          )) // member
-
-public:
-    //!\brief Operator definition.
-    template <typename dummy = int>
-    //!\cond
-        requires requires { { impl(priority_tag<2>{}, s_alph_t{}, dummy{}) }; }
-    //!\endcond
-    constexpr auto operator()() const noexcept
-    {
-        static_assert(noexcept(impl(priority_tag<2>{}, s_alph_t{})),
-            "Only overloads that are marked noexcept are picked up by seqan3::max_pseudoknot_depth.");
-        static_assert(std::constructible_from<size_t, decltype(impl(priority_tag<2>{}, s_alph_t{}))>,
-            "The return type of your max_pseudoknot_depth implementation must be convertible to size_t.");
-        static_assert(SEQAN3_IS_CONSTEXPR(impl(priority_tag<2>{}, s_alph_t{})),
-            "Only overloads that are marked constexpr are picked up by seqan3::max_pseudoknot_depth.");
-
-        return impl(priority_tag<2>{}, s_alph_t{});
-    }
+    //!\brief CRTP base class seqan3::detail::customisation_point_object.
+    using base_t = detail::customisation_point_object<max_pseudoknot_depth_cpo<alphabet_t>, 2>;
+    //!\brief Only this class is allowed to import the constructors from #base_t. (CRTP safety idiom)
+    using base_t::base_t;
+
+    /*!\brief If `alphabet_type` isn't std::is_nothrow_default_constructible, max_pseudoknot_depth will be called with
+     *        std::type_identity instead of a default constructed alphabet.
+     */
+    template <typename alphabet_type>
+    using alphabet_or_type_identity
+        = std::conditional_t<std::is_nothrow_default_constructible_v<std::remove_cvref_t<alphabet_type>> &&
+                             seqan3::is_constexpr_default_constructible_v<std::remove_cvref_t<alphabet_type>>,
+                             std::remove_cvref_t<alphabet_type>,
+                             std::type_identity<alphabet_type>>;
+
+    /*!\brief CPO overload (1. out of 3 checks): explicit customisation via `seqan3::custom::alphabet`
+     * \tparam alphabet_type The type of the alphabet. (Needed to defer instantiation for incomplete types)
+     */
+    template <typename alphabet_type = alphabet_t>
+    static constexpr auto SEQAN3_CPO_OVERLOAD(priority_tag<3>)
```

Comment:
I think we should keep the priority tags conistent?
```suggestion
    static constexpr auto SEQAN3_CPO_OVERLOAD(priority_tag<2>)
```